blog/_posts/2022-04-26-secret-management-the-soft-way.md

```diff
@@ -0,0 +1,212 @@
+The [External Secrets Operator][eso] can connect to a multitude of common secret providers like [Google Secret Manager][gcpsm], [AWS Secrets Manager][awscm] or [Azure Key Vault][azurekv], and provide that data as k8s Secrets.
+To access a secret in Google Cloud, we first need to provide the right credentials. In this case we have created a Service Account in GCP and have downloaded the respective key, which we attached to a Secret that we are passing into a Secret Store.
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: gcpsm-secret
+  labels:
+    type: gcpsm
+type: Opaque
+stringData:
+  secret-access-credentials: |-
+    {
+      "type": "service_account",
+      "project_id": "secret-management-talk",
+      "private_key_id": "",
+      "private_key": "-----BEGIN PRIVATE KEY-----\n\n-----END PRIVATE KEY-----\n",
+      "client_email": "external-secrets@secret-management.iam.gserviceaccount.com",
+      "client_id": "",
+      "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+      "token_uri": "https://oauth2.googleapis.com/token",
+      "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+      "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/external-secrets%40secret-management.iam.gserviceaccount.com"
+    }
+```
+```yaml
+apiVersion: external-secrets.io/v1alpha1
+kind: ClusterSecretStore
+metadata:
+  name: gcp-backend
+spec:
+  provider:
+      gcpsm:                                  # gcpsm provider
+        auth:
+          secretRef:
+            secretAccessKeySecretRef:
+              name: gcpsm-secret              # secret name containing SA key
+              key: secret-access-credentials  # key name containing SA key
+              namespace: eso
+        projectID: secret-management-talk                  # name of Google Cloud project
+```
+
+Now we can tell the operator which data to retrieve and how to provide it to the k8s cluster by defining the target secret in an External Secret resource.
+```yaml
+apiVersion: external-secrets.io/v1alpha1
+kind: ExternalSecret
+metadata:
+  name: gcp-external-secret
+  namespace: gcp-eso-app
+spec:
+  refreshInterval: 1h           # rate SecretManager pulls GCPSM
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: gcp-backend               # name of the SecretStore (or kind specified)
+  target:
+    name: k8s-secret  # name of the k8s Secret to be created
+  data:
+  - secretKey: FOO  # name of the GCPSM secret key
+    remoteRef:
+      key: FOO
+  - secretKey: BAR  # name of the GCPSM secret key
+    remoteRef:
+      key: BAR
+```
+
+
+## HashiCorp Vault & External Secrets Operator
+The External Secrets Operator does not only connect to secret management products provided by the big cloud vendors. If you have bigger security concerns than the mere storing of secrets, you might be using something like [Hashicorp Vault][vault] to control not only your secrets, but also they way they are created and managed.
```

Comment:
i had a hard time parsing this paragraph. the "does not only connect" wording is a bit awkward. "bigger security concerns" is a bit mushy.

Maybe something like:

```
When your team is small, it's OK to throw all your secrets in one bucket. As your team grows, you'll need tools to manage how your secrets are created and who can access them.

Hashicorp Vault can help! 

And the external secrets operator in the last section can connect to Vault too.
```

since it seems like the main things we want to communicate here are (1) why I would use Vault, and (2) how this section relates to the last one